### PR TITLE
Add Tuya Royal Gardineer irrigation valve quirk

### PR DESCRIPTION
**Pull request by mrrstux**
## Proposed change
Quirk for supporting [Royal Gardineer](https://www.royal-gardineer.de/Bewaesserungsventil-ZX-7155-919.shtml) valve.
Similar to this [device](https://www.zigbee2mqtt.io/devices/ZVG1.html).

## Additional information
Exposes:

- open/close command
- battery remaining percentage
- water consumption
- timer time left
- timer state
- last open duration
- weather delay

Valve works on a timer only, with a default timer of 10 minutes. Before opening the valve, it is recommended to set the irrigation desired time in the "Timer time left" field. 

## Checklist
- [x] The changes are tested and work correctly
- [x] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

### CODE REVIEW

**Comment by prairiesnpr** [comment]
Once https://github.com/zigpy/zha-device-handlers/pull/3550 is merged or a variation of it, the correct power config will be available.

File: zhaquirks/tuya/ts0601_valve.py [hunk 1]
```diff
@@ -568,3 +569,74 @@ class GiexIrrigationStatus(t.enum8):
...
+    Disabled = 0x00
+    Active = 0x01
+    Enabled = 0x02
+
+
+(
+    TuyaQuirkBuilder("_TZE200_2wg5qrjy", "TS0601")
+    .tuya_onoff(dp_id=1)
+    # Should be TuyaPowerConfigurationCluster2AA, but it is broken at this time.
+    .tuya_battery(dp_id=7, power_cfg=TuyaPowerConfigurationCluster2AAA)
```
Comment (TheJulianJES) [inline zhaquirks/tuya/ts0601_valve.py]: We should be able to use `TuyaPowerConfigurationCluster2AA` now (fixed in https://github.com/zigpy/zha-device-handlers/pull/3554).
```suggestion
    .tuya_battery(dp_id=7, power_cfg=TuyaPowerConfigurationCluster2AA)
```

**Review by TheJulianJES — APPROVED** [review]
Thanks for the PR! Looks good to me.

I've changed the power configuration cluster to `TuyaPowerConfigurationCluster2AA` now. It should work in Home Assistant Core 2024.12 and later.
It won't work when used as a custom quirk in 2024.11 though.

File: zhaquirks/tuya/ts0601_valve.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
     TUYA_CLUSTER_ID,
     EnchantedDevice,
     TuyaLocalCluster,
+    TuyaPowerConfigurationCluster2AAA,
```
Comment (TheJulianJES) [inline zhaquirks/tuya/ts0601_valve.py:28]:
```suggestion
    TuyaPowerConfigurationCluster2AA,
```

File: zhaquirks/tuya/ts0601_valve.py [hunk 3]
```diff
@@ -568,3 +569,74 @@ class GiexIrrigationStatus(t.enum8):
...
+        attribute_name="timer_time_left",
+        type=t.uint32_t,
+        min_value=1,
+        max_value=600,
+        step=1,
+        multiplier=1 / 60,
+        unit=UnitOfTime.MINUTES,
+        translation_key="timer_time_left",
+        fallback_name="Timer time left",
+    )
```
Comment (TheJulianJES) [inline zhaquirks/tuya/ts0601_valve.py:610]: @mrrstux Just to confirm, this is adjustable by the user? Or should it be a sensor?

File: zhaquirks/tuya/ts0601_valve.py [hunk 4]
```diff
@@ -568,3 +569,74 @@ class GiexIrrigationStatus(t.enum8):
...
+
+    Disabled = 0x00
+    Active = 0x01
+    Enabled = 0x02
+
+
+(
+    TuyaQuirkBuilder("_TZE200_2wg5qrjy", "TS0601")
+    .tuya_onoff(dp_id=1)
+    # Should be TuyaPowerConfigurationCluster2AA, but it is broken at this time.
```
Comment (TheJulianJES) [inline zhaquirks/tuya/ts0601_valve.py:594]: Hmm, I should have removed that comment. I'll address it in a follow-up PR.

File: zhaquirks/tuya/ts0601_valve.py (hunk 3 above)
Comment (TheJulianJES) [reply zhaquirks/tuya/ts0601_valve.py:610]: Ok, looking at Z2M, that this value is adjustable. It just has two "exposes": one only readable, one only writeable.

File: zhaquirks/tuya/ts0601_valve.py (hunk 4 above)
Comment (TheJulianJES) [reply zhaquirks/tuya/ts0601_valve.py:594]: https://github.com/zigpy/zha-device-handlers/pull/3555

**Pull request merged**